pod.go

```diff
@@ -774,6 +776,45 @@ func (p *Pod) startVM(netNsPath string) error {
 	return nil
 }
 
+func (p *Pod) createProxyProcess() error {
+	pid, uri, err := p.proxy.start(*p)
+	if err != nil {
+		return err
+	}
+
+	// save state
+	p.state.URL = uri
+	if err := p.setPodState(p.state); err != nil {
+		return err
+	}
+
+	p.proxyProcess = newProcess("", pid)
+
+	p.Logger().WithField("proxy-pid", pid).Debug("proxy started")
+
+	return nil
+}
+
+// waitForProxy waits for the proxy process to finish
+func (p *Pod) waitForProxy() error {
+	pid := p.proxyProcess.Pid
+	if pid == 0 {
+		return nil
+	}
+
+	// cannot fail on Linux
+	proc, _ := os.FindProcess(pid)
+
+	_, err := proc.Wait()
```

Comment:
We should add a timeout here.

Reply:
Good call.

Looking at this again, the code actually won't work as we can't guarantee the calling process (now) owns the `cc-proxy` instance (it's likely to have been reparented to PID 1).

Do we already have BKM code somewhere for a (deep breath) pid or`/proc/$pid`-sniffing loop? I'm pretty sure we used to a long time ago but can't find it atm.